storage/flux/table.gen.go.tmpl

```diff
@@ -299,6 +300,370 @@ func (t *{{.name}}WindowTable) advance() bool {
 	return true
 }
 
+// window selector table
+type {{.name}}WindowSelectorTable struct {
+	{{.name}}Table
+	windowEvery int64
+	timeColumn  string
+}
+
+func new{{.Name}}WindowSelectorTable(
```

Comment:
I think it might be possible for us to integrate this code with the existing window code although we might need to pass in a selector flag. I'd like to pair on it, but here's the process for how the current window works.

1. createNextBufferTimes which creates the start and stop columns for each window. In the aggregate window version, one of these is taken to become the time column.
2. merge the values from the buffer to match with the windows that were just calculated.

I am wondering if it's possible for us to merge the selector into this. A lot of the logic is still the same. We still need to compute the start and stop times and we still need to merge the values from the the array cursor. I think we should give it a try.

Reply:
I'm not opposed to it, but I'd prefer if they were separated. While we still need to compute start and stop times, the time stamps are not interpreted in the same way as they are for aggregates. Combining the logic I think is going to lead to confusion.

Reply:
The array that other aggregates return has the window stop time in their timestamps, but selectors return the selected rows real timestamp, right?

Reply:
That's correct.

Reply:
I kinda believe that it is OK for the selector tables to stand out but if @jsternberg would love to give it a try to merge them I am also interested to see his changes.

Reply:
We've decided to hold off on merging the implementations at this point. While there is some overlapping logic between the two, because there are a decent amount of edge cases to account for during windowing, a separate selector implementation keeps branching to a minimum which makes it much easier to follow the algorithm.